File: python/Human.py
class Human:
    def __init__(self, nama='', nik=0, kelamin='-'):
        self.nama = nama
        self.nik = nik
        self.kelamin = kelamin
    
    def get_nama(self):
        return self.nama
    
    def set_nama(self, nama):
        self.nama = nama
    
    def get_nik(self):
        return self.nik
    
    def set_nik(self, nik):
        self.nik = nik
    
    def get_kelamin(self):
        return self.kelamin
    
    def set_kelamin(self, kelamin):
        self.kelamin = kelamin
    
    def __del__(self):
        pass


class Mahasiswa(Human):
    def __init__(self, nama='', nik=0, kelamin='-', nim=0, prodi='-', fakultas='-'):
        super().__init__(nama, nik, kelamin)
        self.nim = nim
        self.prodi = prodi
        self.fakultas = fakultas
    
    def get_nim(self):
        return self.nim
    
    def set_nim(self, nim):
        self.nim = nim
    
    def get_prodi(self):
        return self.prodi
    
    def set_prodi(self, prodi):
        self.prodi = prodi
    
    def get_fakultas(self):
        return self.fakultas
    
    def set_fakultas(self, fakultas):
        self.fakultas = fakultas
    
    def __del__(self):
        pass


class SivitasAkademik(Mahasiswa):
    def __init__(self, nama='', nik=0, kelamin='-', nim=0, prodi='-', fakultas='-', asal_univ='-', email='-'):
        super().__init__(nama, nik, kelamin, nim, prodi, fakultas)
        self.asal_univ = asal_univ
        self.email = email
    
    def get_univ(self):
        return self.asal_univ
    
    def set_univ(self, univ):
        self.asal_univ = univ
    
    def get_email(self):
        return self.email
    
    def set_email(self, email):
        self.email = email
    
    def __del__(self):
        pass

File: new_java/Human.java
package new_java;

class Human {
    protected String nama;
    protected long nik;
    protected String kelamin;

    public Human() {
        this.nama = "";
        this.nik = 0;
        this.kelamin = "-";
    }

    public Human(String nama, long nik, String kelamin) {
        this.nama = nama;
        this.nik = nik;
        this.kelamin = kelamin;
    }

    public String getNama() {
        return this.nama;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public long getNik() {
        return this.nik;
    }

    public void setNik(long nik) {
        this.nik = nik;
    }

    public String getKelamin() {
        return this.kelamin;
    }

    public void setKelamin(String kelamin) {
        this.kelamin = kelamin;
    }
}

class Mahasiswa extends Human {
    private long nim;
    private String prodi;
    private String fakultas;

    public Mahasiswa() {
        this.nim = 0;
        this.prodi = "-";
        this.fakultas = "-";
    }

    public Mahasiswa(String nama, long nik, String kelamin, long nim, String prodi, String fakultas) {
        super(nama, nik, kelamin);
        this.nim = nim;
        this.prodi = prodi;
        this.fakultas = fakultas;
    }

    public long getNim() {
        return this.nim;
    }

    public void setNim(long nim) {
        this.nim = nim;
    }

    public String getProdi() {
        return this.prodi;
    }

    public void setProdi(String prodi) {
        this.prodi = prodi;
    }

    public String getFakultas() {
        return this.fakultas;
    }

    public void setFakultas(String fakultas) {
        this.fakultas = fakultas;
    }
}

class SivitasAkademik extends Mahasiswa {
    private String asalUniv;
    private String email;

    public SivitasAkademik() {
        this.asalUniv = "-";
        this.email = "-";
    }

    public SivitasAkademik(String nama, long nik, String kelamin, long nim, String prodi, String fakultas,
            String asalUniv, String email) {
        super(nama, nik, kelamin, nim, prodi, fakultas);
        this.asalUniv = asalUniv;
        this.email = email;
    }

    public String getAsalUniv() {
        return this.asalUniv;
    }

    public void setAsalUniv(String asalUniv) {
        this.asalUniv = asalUniv;
    }

    public String getEmail() {
        return this.email;
    }

    public void setEmail(String email) {
        this.email = email;
    }
}
